include/ensmallen_bits/adamwr/adamwr.hpp

```diff
@@ -0,0 +1,155 @@
+/**
+ * @file adamwr.hpp
+ * @author Niteya Shah
+ * Declaration of the AdamWR optimizer.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef ENSMALLEN_ADAM_ADAMWR_HPP
+#define ENSMALLEN_ADAM_ADAMWR_HPP
+
+#include <ensmallen_bits/adamw/adamw_update.hpp>
+#include <ensmallen_bits/sgdr/cyclical_decay.hpp>
+#include <ensmallen_bits/sgd/sgd.hpp>
+
+namespace ens {
+/*
+ *
+ * @code
+ * @article{
+ *   title   = {Decoupled Weight Decay Regularization},
+ *   author  = {Loschilov, I. and Hutter, F.},
+ *   journal = {ArXiv e-prints},
+ *   url     = {https://arxiv.org/pdf/1711.05101.pdf}
+ *   year    = {2019}
+ * }
+ * @endcode
+ *
+ * AdamWR can optimize differentiable separable functions.  For more details, see
+ * the documentation on function types included with this distribution or on the
+ * ensmallen website.
+*/
+class AdamWR
+{
+ public:
+  /**
+   * Construct the AdamWR optimizer with the given function and parameters.The
+   * maximum number of iterations refers to the maximum number of points that
+   * are processed (i.e., one iteration equals one point; one iteration does not
+   * equal one pass over the dataset).
+   *
+   * @param stepSizeMax Maximum and initial step size for each batch of warm
+     restart.
+   * @param stepSizeMin Minimum and final step size for each batch of warm
+     restart.
+   * @param epochRestart Restart Rate for Warm Restarts
+   * @param multFactor Multiplier for epochRestart
+   * @param weightDecay The rate at which the update regularises the iterate.
+   * @param batchSize Number of points to process in a single step.
+   * @param beta1 Exponential decay rate for the first moment estimates.
+   * @param beta2 Exponential decay rate for the weighted infinity norm
+                  estimates.
+   * @param eps Value used to initialise the mean squared gradient parameter.
+   * @param maxIterations Maximum number of iterations allowed (0 means no
+   *                      limit).
+   * @param tolerance Maximum absolute tolerance to terminate algorithm.
+   * @param shuffle If true, the function order is shuffled; otherwise, each
+   *        function is visited in linear order.
+   * @param resetPolicy If true, parameters are reset before every Optimize
+   *        call; otherwise, their values are retained.
+   * @param updateRule Update Policy to be used for AdamR.
+   * @param decayPolicy Decay Policy to be used for AdamR.
+   */
+  AdamWR(const double stepSizeMax = 0.002,
```

Comment:
Looks like we can just use the ``AdamRType`` class here instead of adding ``AdamWR``  and add another alias?

Reply:
The problem here is the same as with AdamW . AdamWR has an extra parameter of weight Decay and I dont want to overload the AdamR constructor . 
Please correct me if I misunderstood your doubt.

Reply:
Ahh, you are right I missed that.